kolibri/core/notifications/test/test_api.py

```diff
@@ -168,22 +200,232 @@ def test_create_notification(self):
+        self.summarylog1.progress = 1.0
+        self.summarylog1.completion_timestamp = local_now() + timedelta(seconds=2)
+        self.summarylog1.end_timestamp = local_now() + timedelta(seconds=5)
+        self.summarylog1.save()
+        masteryLog_1 = MasteryLogFactory.create(
+            summarylog=self.summarylog1,
+            start_timestamp=self.summarylog1.start_timestamp,
+            completion_timestamp=self.summarylog1.completion_timestamp,
+            end_timestamp=self.summarylog1.end_timestamp,
+            user=self.user1,
+            complete=True,
+        )
+
+        parse_summarylog(self.summarylog1)
+
+        # User tried again
+        self.summarylog1.progress = 1.0
+        self.summarylog1.completion_timestamp = local_now() + timedelta(seconds=2)
+        self.summarylog1.end_timestamp = local_now() + timedelta(seconds=5)
+        self.summarylog1.save()
+        MasteryLogFactory.create(
+            summarylog=self.summarylog1,
+            start_timestamp=self.summarylog1.start_timestamp,
+            completion_timestamp=self.summarylog1.completion_timestamp,
+            end_timestamp=self.summarylog1.end_timestamp,
+            user=self.user1,
+            complete=True,
+        )
+
+        parse_summarylog(self.summarylog1)
+
+        # Only one notification should be created with the first masterylog completion timestamp
+        notifications = LearnerProgressNotification.objects.filter(
+            notification_object=NotificationObjectType.Resource,
+            notification_event=NotificationEventType.Completed,
+            user_id=self.user1.id,
+            classroom_id=self.classroom.id,
+            assignment_collections=[self.classroom.id],
+            contentnode_id=self.node_1.id,
+            lesson_id=self.lesson.id,
+        )
+        assert notifications.count() == 1
+        assert notifications[0].timestamp == masteryLog_1.completion_timestamp
+
+    def test_parse_retry_summarylog_dont_update_lesson_completed_notification(self):
+        self.summarylog1.progress = 1.0
+        self.summarylog1.completion_timestamp = local_now() + timedelta(seconds=2)
+        self.summarylog1.end_timestamp = local_now() + timedelta(seconds=5)
+        self.summarylog1.save()
+        MasteryLogFactory.create(
+            summarylog=self.summarylog1,
+            start_timestamp=self.summarylog1.start_timestamp,
+            completion_timestamp=self.summarylog1.completion_timestamp,
+            end_timestamp=self.summarylog1.end_timestamp,
+            user=self.user1,
+            complete=True,
+        )
+
+        parse_summarylog(self.summarylog1)
+
+        self.summarylog2.progress = 1.0
+        self.summarylog2.completion_timestamp = local_now() + timedelta(seconds=2)
+        self.summarylog2.end_timestamp = local_now() + timedelta(seconds=5)
+        self.summarylog2.save()
+        masteryLog_2 = MasteryLogFactory.create(
```

Comment:
This variable name is mildly confusing - as calling it `masteryLog_2` makes it feel like it's the second masterylog for something (although I see it is meant to be the masterylog for summarylog2).

Maybe just keeping the casing consistent with the above would help, so `masterylog2` to parallel `sumamrylog2` (that would also remove the little diversion into camelCase too :) )

Reply:
Done!